pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -58,44 +65,56 @@ func (s *Store) Signatures(ctx context.Context, name string, digest string, fn s
 		return done, err
 	}
 
-	stores := make([]store.Store, 0, len(uris))
-	for i := range uris {
-		uri := *uris[i]
-		stores = append(stores, &sigstore.Store{
-			URI:        &uri,
-			HTTPClient: s.HTTPClient,
-		})
+	stores := make([]store.Store, 0, len(customStores))
+	for _, customStore := range customStores {
+		uri, err := url.Parse(customStore.URL)
+		if err != nil {
+			return err
+		}
+		newHttpClient, err := s.HTTPClient(customStore.CA.Name)
+		if err != nil {
+			return err
+		}
+		if customStore.CA.Name != "" {
+			stores = append(stores, &sigstore.Store{
+				URI:        uri,
+				HTTPClient: func() (*http.Client, error) { return newHttpClient, nil }})
+		}
+
+	}
+	if len(stores) == 0 {
+		return errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature")
```

Comment:
I don't think the message is valid at this point - we have not even started to look for signatures (that's done by `Signatures`  call on L89). We just filtered out CAs without a name and were left at zero

Reply:
Updated the message as `ClusterVersion spec.signatureStores exhausted without finding a valid signature store`

Reply:
Actually removed the error message from here and moved this to the end of the method.

Reply:
related error-handling discussion [in this new thread][1] that addresses Petr's concern, and as of 4ead92af5e09, this error message is back at the end of the `Store.Signatures` function where it belongs, so I'm marking this thread resolved.

[1]: https://github.com/openshift/cluster-version-operator/pull/1030#discussion_r1563333651